### PR TITLE
[CI] Deprecate sdk_runner and prod_v1 from OSS release tests

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
That time has come. There are now no remaining OSS release tests using another type runner or another anyscale stack. Everything now runs on staging v2 through anyscale job submission.

## Checks
- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests

### CODE REVIEW

**Comment by can-anyscale** [comment]
@architkulkarni: hopefully this doesn't make your work in creating similar tests in GCE harder; i know a thing or two on how to make these tests run now on anyscale so let me know if there are blockers i can help

**Review by aslonnie — APPROVED** [review]

File: python/ray/autoscaler/aws/tests/aws_compute.yaml
Comment (ijrsvt) [inline python/ray/autoscaler/aws/tests/aws_compute.yaml:1]: qq, where is this specified? Or is that up to the test runner to specify this?

**Review by architkulkarni — APPROVED** [review]

**Comment by architkulkarni** [comment]
Assigning @wuisawesome to unblock codeowner review.

File: python/ray/autoscaler/aws/tests/aws_compute.yaml
Comment (can-anyscale) [reply python/ray/autoscaler/aws/tests/aws_compute.yaml:1]: totally, it's define in an .env file https://github.com/ray-project/ray/blob/master/release/ray_release/environments/staging_v2.env#L3

**Review by DmitriGekhtman — APPROVED** [review]
[stamp]

**Review by aslonnie — APPROVED** [review]

**Review by Yard1 — APPROVED** [review]

**Pull request merged**
